cmd/prometheus/main.go

```diff
@@ -208,6 +208,12 @@ func (c *flagConfig) setFeatureListOptions(logger log.Logger) error {
 			}
 		}
 	}
+
+	if c.tsdb.EnableNativeHistograms && c.tsdb.EnableMemorySnapshotOnShutdown {
+		c.tsdb.EnableMemorySnapshotOnShutdown = false
+		level.Warn(logger).Log("msg", "memory-snapshot-on-shutdown feature does not support native-histograms. Hence, memory-snapshot-on-shutdown is disabled automatically since native-histograms is enabled.")
```

Comment:
"Hence" and "since" in one sentence sounds somewhat redundant. How about this:

`"memory-snapshot-on-shutdown has been disabled automatically because memory-snapshot-on-shutdown and native-histograms cannot be enabled at the same time."`